openedx/core/djangoapps/course_groups/management/commands/tests/test_remove_users_from_multiple_cohorts.py

```diff
@@ -0,0 +1,70 @@
+"""
+from django.core.exceptions import MultipleObjectsReturned
+from django.core.management import call_command
+from django.test.client import RequestFactory
+
+from openedx.core.djangoapps.course_groups.views import cohort_handler
+from openedx.core.djangoapps.course_groups.cohorts import get_cohort, get_cohort_by_name
+from openedx.core.djangoapps.course_groups.tests.helpers import config_course_cohorts
+from student.tests.factories import UserFactory
+from xmodule.modulestore.tests.django_utils import ModuleStoreTestCase
+from xmodule.modulestore.tests.factories import CourseFactory
+
+
+class TestMultipleCohortUsers(ModuleStoreTestCase):
+    """
+    Base class for testing users with multiple cohorts
+    """
+    def setUp(self):
+        """
+        setup course, user and request for tests
+        """
+        super(TestMultipleCohortUsers, self).setUp()
+        self.course = CourseFactory.create()
+        self.user1 = UserFactory(is_staff=True)
+        self.user2 = UserFactory(is_staff=True)
+        self.request = RequestFactory().get("dummy_url")
+        self.request.user = self.user1
+
+    def test_users_with_multiple_cohorts_cleanup(self):
+        """
+        Test that user which have been added in multiple cohorts of a course,
+        can get cohorts without error after running cohorts cleanup command
+        """
+        # set two auto_cohort_groups
+        config_course_cohorts(
+            self.course, [], cohorted=True, auto_cohort_groups=["AutoGroup1", "AutoGroup2"]
+        )
+
+        # get the cohorts from the course, which will cause auto cohorts to be created.
+        cohort_handler(self.request, unicode(self.course.id))
+        auto_cohort_1 = get_cohort_by_name(self.course.id, "AutoGroup1")
+        auto_cohort_2 = get_cohort_by_name(self.course.id, "AutoGroup2")
+
+        # forcefully add user1 in two auto cohorts
+        auto_cohort_1.users.add(self.user1)
+        auto_cohort_2.users.add(self.user1)
+        # forcefully add user2 in an auto cohort too
+        auto_cohort_1.users.add(self.user2)
+
+        # now check that when user1 goes on discussion page and tries to get
+        # cohorts 'MultipleObjectsReturned' exception is returned
+        with self.assertRaises(MultipleObjectsReturned):
+            get_cohort(self.user1, self.course.id)
+        # also check that user 2 can go on discussion page without any exception
+        get_cohort(self.user2, self.course.id)
+
+        # call command to remove users added in multiple cohorts of a course
+        # are removed from all cohort groups
+        call_command('remove_users_from_multiple_cohorts')
+
+        # check that only user1 (with multiple cohorts) is removed from cohort
+        # and user2 still in auto cohort after running management command
+        self.assertEqual(self.user1.course_groups.count(), 0)
+        self.assertEqual(self.user2.course_groups.count(), 1)
```

Comment:
@zubair-arbi Please also create a second course with cohorts and add `user2` to one of them so that `user2` is in 2 cohorts. At this point they should still be in cohorts in both the courses.